### PR TITLE
Normative: Initial spec draft

### PR DESCRIPTION
**Pull request by syg**

### CODE REVIEW

**Comment by rbuckton** [comment]
This would be a bit easier to review with a generated preview. I usually use this workflow, which publishes to the `gh-pages` branch: https://github.com/tc39/proposal-extractors/blob/main/.github/workflows/pr.yml

File: spec/index.html [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+<pre class="metadata">
+title: JavaScript Structs
+stage: 1
+contributors: Shu-yu Guo
+markEffects: true
+</pre>
+
+<emu-intro id="intro">
+  <h1>Structs, Shared Structs, Unsafe Blocks, and Synchronization Primitives</h1>
+  <p>We extend the JS language with fixed-layout objects both for unshared and shared uses, unsafe blocks to syntactically to limit the scope of where racy accesses may be performed, and high-level synchronization primitive APIs. This spec draft is organized by logical feature.</p>
```
Comment (acutmore) [inline spec/index.html:12]:
```suggestion
  <p>We extend the JS language with fixed-layout objects both for unshared and shared uses, unsafe blocks to syntactically limit the scope of where racy accesses may be performed, and high-level synchronization primitive APIs. This spec draft is organized by logical feature.</p>
```

File: spec/index.html [hunk 2]
```diff
@@ -0,0 +1,1272 @@
...
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-shared-structs">
+  <h1>Shared Structs</h1>
+
+  <emu-clause id="sec-shared-struct-exotic-objects">
+    <h1>Shared Struct Exotic Objects</h1>
+    <ins class="block">
+    <p>Shared Structs are fixed-layout exotic objects that can be shared across agents and be accessed in parallel from multiple agents. They are like structs with more restricted behaviour so as to be possible to be shared across agents. They cannot contain methods or private fields. Their fields can only hold primitives or other shared values. Accessing their fields is unordered by default and is governed by the memory model. Such accesses can be made sequentially consistent by using newly overloaded Atomics methods.</p>
```
Comment (Jamesernator) [inline spec/index.html:501]: > They cannot contain methods or private fields.

Given shared structs will be able to be correlated by source position anyway is the restriction on private fields really neccessary? It seems unfortunate that things like shared queues can't encapsulate their internals.

Comment (syg) [reply spec/index.html:501]: Agreed, that'd be nice, but this proposal is already quite large, so one step at a time. If the correlation registry thing gets consensus, private fields is a sensible immediate follow-up proposal.

**Pull request merged**